core/rawdb/accessors_chain.go

```diff
@@ -829,12 +1010,33 @@ func writeAncientBlock(op ethdb.AncientWriteOp, block *types.Block, header *type
 	return nil
 }
 
+func writeAncientSidecar(op ethdb.AncientWriteOp, num uint64, blobs types.BlobTxSidecars) error {
+	if err := op.Append(ChainFreezerBlobSidecarTable, num, blobs); err != nil {
```

Comment:
can we move the logic into func `writeAncientBlock`?
check len(block.Sidecars()) firstly, and by this, we can remove writeAncientBlockWithSidecar
and simplify WriteAncientBlocksAfterCancun